### DISCUSSION TITLE
Scroll intertia carries over from TUI application to scrollback buffer

### DISCUSSION THREAD

**Discussion by ComplexPlane** [ghostty-org/ghostty#3838] [Feature Requests, Ideas] [answered]
I noticed that when using a Macbook trackpad's inertial scrolling inside a TUI application, if a scroll is started and then the app is quit right after, the scroll inertia carries over to the scrollback buffer. This happens specifically for me when scrolling up (fingers moving down) in a Lazygit window and then hitting `q`. This doesn't seem to happen in Kitty so I guess it's preventable?

## Ghostty

https://github.com/user-attachments/assets/351a4625-20d5-4579-bf26-afbc624d5e9e

## Kitty

https://github.com/user-attachments/assets/98a37fa7-5d48-4170-b812-9a6ec59d2297

**Comment by mitchellh** [comment, marked as answer]
Accepted: #3838

**Comment by gigamonster256** [comment]
When I'm scrolling through the process list in btop and exit while the inertial scroll is still going, I get repeating sequences such as: 144;30M64; or 129;32M65; (the sequence printed many times over until the inertial scroll stops) in addition to the behavior observed above.
